src/talk/errors.rs

```diff
@@ -0,0 +1,42 @@
+use err_derive::Error;
+use gstreamer::glib;
+
+/// The main error for the status-light subcommand
```

Comment:
I think this needs to be coalesced into a single Error type, as it seems prone to duplication.

However, this is at the top application level. I particularly like the `anyhow` crate for application top level errors like this. Perhaps using it would get rid of the giant enum altogether. We don't really care about the API surface, we just want the application to crash :)

Reply:
Sure I think, errors in the library should be formally defined but outside in application level `anyhow` should be good

Reply:
I've been reading up on error libraries [here](https://blog.yoshuawuyts.com/error-handling-survey/) and learned about thiserror and other libraries. It made we wonder if there was a specific motivation for using err-derive over others?

Reply:
I've gone and use anyhow and this removed the need for the error type at the application level